execute/observation.go

```diff
@@ -454,6 +454,10 @@ func (p *Plugin) getMessagesObservation(
 			}
 		}
 
+		// TODO: check if the commit report is all fully executed and inflight messages.
+		// Might be able to just not include it in the observation at all? Or is there a
+		// risk with that?
```

Comment:
@0xnogo @winder thoughts on this? Does it make sense?

Reply:
Yeah that makes sense. There's code in here somewhere that attempts to remove fully executed commit reports. Maybe the inflight cache should use seqNum so that we can include it in the "executed message" list to trigger that code.

Reply:
Follow up: https://smartcontract-it.atlassian.net/browse/CCIP-5816